modules/php/Models/Scoresheets/Scoresheet2.php

```diff
@@ -169,12 +169,10 @@ public function canUseJoker(): bool
 
   public function getFirstUnscribbledJoker(): int|null
   {
-    foreach (CircleOther::$jokers as $jokerSlot) {
-      if ($this->hasScribbledSlot($jokerSlot, SCRIBBLE_CIRCLE) && !$this->hasScribbledSlot($jokerSlot, SCRIBBLE)) {
-        return $jokerSlot;
-      }
-    }
-    return null;
+    $circledJokers = array_filter(CircleOther::$jokers, function ($jokerSlot) {
```

Comment:
Arrow function ? :)

Reply:
Actually, maybe a generic method that filter an array of slots with respect to a slot type ?

Reply:
But that's what I've done: getFirstUnscribbled() in Scoresheet.php. Or did you mean something else?

Reply:
Exactly this, but instead of returning the first, just return all of them that are not scribbled over.
(Obviously you would move the logic from getFirstUnscribbled into this new function and just call this function and return the first element)

Reply:
Will do this in the next PR then :)